JustSaying/AwsTools/MessageHandling/SnsTopicBase.cs

```diff
@@ -67,16 +70,27 @@ public void Publish(Message message)
             try
             {
                 await Client.PublishAsync(request);
+
                 _eventLog.LogInformation($"Published message: '{request.Subject}' with content {request.Message}");
             }
             catch (Exception ex)
             {
-                throw new PublishException(
-                    $"Failed to publish message to SNS. TopicArn: {request.TopicArn} Subject: {request.Subject} Message: {request.Message}",
-                    ex);
+                if (!ClientExceptionHandler(ex, request))
+                    throw new PublishException(
+                        $"Failed to publish message to SNS. TopicArn: {request.TopicArn} Subject: {request.Subject} Message: {request.Message}",
+                        ex);
             }
         }
 
+        private bool ClientExceptionHandler(Exception ex, PublishRequest request)
+        {
+            bool exceptionIsHandled = false;
+            if (SnsWriteConfiguration?.OnException != null)
+                exceptionIsHandled = SnsWriteConfiguration.OnException(ex, request);
```

Comment:
Could use `?.Invoke...` for a single liner

Reply:
👍